### DISCUSSION TITLE
Widget Vue components as npm package

### DISCUSSION THREAD

**Discussion by jorgevrgs** [PrestaShop/PrestaShop#33412] [Ideas]
#### Is your feature request related to a problem?

The New theme includes the folder `https://github.com/PrestaShop/PrestaShop/tree/develop/admin-dev/themes/new-theme/js/app/widgets` with components available in local files.

#### Describe the solution you'd like

Deliver the Vue components as npm package like `npm install @prestashop/vue-components` instead of a local folder to allow modules to reuse these components.

#### Alternatives you've considered

Generate an npm package with these components and include it in the current controllers using Vue like `import { PSTable } from "@prestashop/vue-components"` to be reusable.

#### Additional context

NA.

**Comment by hibatallahAouadni** [comment]
Hello @jorgevrgs 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by matks** [comment]
ping @NeOMakinG 

@jorgevrgs what would be the benefits of doing this, in comparison to current implementation?

**Comment by NeOMakinG** [comment]
@matks I already thought about this when we had the blockwishlist hard path problem

I would love to, but it add a new step to the release of PS, and it could cause missversion issues, this is something we should discuss with the team in order to list pros and cons

**Comment by jorgevrgs** [comment]
> ping @NeOMakinG
> 
> @jorgevrgs what would be the benefits of doing this, in comparison to current implementation?

Mainly code reuse for other controllers that may require it, not only stock controller, and allow developers to use it in their modules. 

It could be a `prestakit-ui-kit` implementation for Vue, a kind of `@prestashop/prestakit-ui-vue` package, being ambitious, opening for a (not-too-far) future back-office API-based without reloading (decoupled architecture), furthermore to standardize the use and generate best practices documentation.
